src/main/java/xyz/niflheim/stockfish/engine/util/Util.java

```diff
@@ -0,0 +1,47 @@
+package xyz.niflheim.stockfish.engine.util;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Comparator;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class Util {
```

Comment:
I propose to call this class FileEngineUtil so that in the future other code that does not relate to files will not appear in this class. In general, the Util class is a bad practice, as it usually turns into a dump; it is better to make an interface for string constants.
But I propose to make this a class and move here the logic responsible for creating the path, then it will be possible to use everything, preferably TreeMap, such as TreeMap, instead of iterators and stream. I think this will make the code more elegant.

Reply:
Implemented as you requested